File: module-git/src/main/java/com/timecat/module/git/sgit/activities/delegate/actions/DiffAction.java
package com.timecat.module.git.sgit.activities.delegate.actions;

import com.timecat.module.git.sgit.activities.RepoDetailActivity;
import com.timecat.module.git.sgit.database.models.Repo;

public class DiffAction extends RepoAction {

    public DiffAction(Repo repo, RepoDetailActivity activity) {
        super(repo, activity);
    }

    @Override
    public void execute() {
        mActivity.enterDiffActionMode();
        mActivity.closeOperationDrawer();
    }
}

File: module-git/src/main/java/com/timecat/module/git/sgit/activities/delegate/actions/DiffAction.kt
package com.timecat.module.git.sgit.activities.delegate.actions

import com.timecat.module.git.sgit.activities.RepoDetailActivity
import com.timecat.module.git.sgit.database.models.Repo

class DiffAction(repo: Repo, activity: RepoDetailActivity) : RepoAction(repo, activity) {
    override fun execute() {
        mActivity.enterDiffActionMode()
    }
}